### PR TITLE
fixes #19543 - act key repo sets toggles

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html [hunk 1]
```diff
@@ -16,6 +16,18 @@
            ng-model="repositorySetFilter"/>
   </div>
 
+  <div data-block="filters">
+    <label class="checkbox-inline" title="{{ 'Show all Repository Sets available to this Activation Key' | translate }}">
+      <input type="checkbox" ng-model="contentAccessModes.contentAccessModeAll" ng-change="toggleFilters()"/>
+      <span translate>Show All</span>
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html]: When testing this it wasn't clear to me what "Show All" meant.  Does this mean show all that aren't in the CVE?

Comment (thomasmckay) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html]: Is the help tip not descriptive enough? I'm open to better wording.

Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html]: You know, I didn't even notice the tooltip.  When you say "available" what does that mean?  I think that is my confusion.

Comment (thomasmckay) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html]: "Show all Repository Sets currently available for Organization, not just those available for this activation key through its subscriptions."

How about that? That will match the info tip always displayed on page.

File: app/models/katello/activation_key.rb [hunk 2]
```diff
@@ -99,11 +99,32 @@ def products
...
 
     def available_content
-      self.products ? self.products.map(&:available_content).flatten.uniq { |product| product.content.id } : []
+      self.products.map(&:available_content).flatten.uniq { |product| product.content.id }
+    end
+
+    def available_product_content(content_access_mode_all = false, content_access_mode_env = false)
+      if content_access_mode_env && environment_id && content_view_id
+        version = ContentViewVersion.in_environment(environment_id).where(:content_view_id => content_view_id).first
+        content_view_version_id = version.id
```
Comment (parthaa) [inline app/models/katello/activation_key.rb]: might need to handle the case where AK has a CV but nothing published yet.. Version likely to be nil

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html (hunk 1 above)
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/views/activation-key-repository-sets.html]: Haha, that's a little wordy.  What about just "Show all Repository Sets in the Organization"?

File: app/models/katello/activation_key.rb (hunk 2 above)
Comment (jlsherrill) [inline app/models/katello/activation_key.rb]: You can't set a CV/LE on a key if its not published to that env, however always good to handle that case !

Comment (thomasmckay) [inline app/models/katello/activation_key.rb]: @waldenraines @parthaa - updated

File: app/models/katello/activation_key.rb [hunk 3]
```diff
@@ -99,11 +99,33 @@ def products
...
+    def all_products
+      Katello::Product.joins(:subscriptions => :pools).where(:organization_id => organization.id).enabled.uniq
     end
 
     def available_content
-      self.products ? self.products.map(&:available_content).flatten.uniq { |product| product.content.id } : []
+      self.products.map(&:available_content).flatten.uniq { |product| product.content.id }
+    end
+
+    def available_product_content(content_access_mode_all = false, content_access_mode_env = false)
```
Comment (jlsherrill) [inline app/models/katello/activation_key.rb]: confused about these two methods 'available_content' and 'available_product_content', i see they do somewhat different things, but are they both needed?

File: app/models/katello/product.rb [hunk 4]
```diff
@@ -212,9 +212,10 @@ def syncable_content?
       repositories.any?(&:url?)
     end
 
-    def available_content
+    def available_content(content_view_version_id = nil)
       self.productContent.select do |content|
-        self.repositories.subscribable.where(content_id: content.content.id).exists?
+        repos = self.repositories.subscribable.where(content_id: content.content.id)
+        repos.exists? && (content_view_version_id.nil? || repos.where(content_view_version_id: content_view_version_id.to_i).count > 0)
```
Comment (jlsherrill) [inline app/models/katello/product.rb]: wouldn't putting repos.exists? here cause it to return false if there were none? (rather than an empty array)?

Comment (parthaa) [inline app/models/katello/product.rb]: @jlsherrill confused. Its inside a select block and expects a boolean right ?

**Comment by parthaa** [comment]
lgtm

File: app/models/katello/product.rb (hunk 4 above)
Comment (jlsherrill) [inline app/models/katello/product.rb]: oh right, nevermind! carry on!

File: app/models/katello/activation_key.rb (hunk 3 above)
Comment (thomasmckay) [inline app/models/katello/activation_key.rb]: @jlsherrill made a single method and added uniq to returned results

**Comment by jlsherrill** [comment]
I found one small problem:

Create an activation key
Click on 'repository sets'
Click on 'Show all' (notice its showing all)
Navigate to the subscription tab
Navigate back to the repository sets tab
Notice that its still 'showing all', but the 'show all' checkbox is not checked. 

Its not a huge deal, but i could see it being confusing (it confused me)

**Comment by thomasmckay** [comment]
@waldenraines see @jlsherrill comment above, not sure how to reset the vars (and refresh data?) when returning to the page

**Comment by waldenraines** [comment]
> @waldenraines see @jlsherrill comment above, not sure how to reset the vars (and refresh data?) when returning to the page

That seems to happen on other pages (errata, packages, etc.) with this pattern as well.  Can one of y'all please file a bug and I'll take a look at it?  Thanks!

**Comment by thomasmckay** [comment]
@waldenraines @jlsherrill http://projects.theforeman.org/issues/19619

**Comment by jlsherrill** [comment]
ACk from me then

**Pull request merged**